driver-core/src/main/java/com/datastax/driver/core/TabletMapListener.java

```diff
@@ -0,0 +1,29 @@
+package com.datastax.driver.core;
+
+public class TabletMapListener extends SchemaChangeListenerBase {
+  private final TabletMap tabletMap;
+
+  public TabletMapListener(TabletMap tabletMap) {
+    this.tabletMap = tabletMap;
+  }
+
+  @Override
+  public void onTableChanged(TableMetadata current, TableMetadata previous) {
+    tabletMap.removeTableMappings(previous.getKeyspace().getName(), previous.getName());
+  }
+
+  @Override
+  public void onTableRemoved(TableMetadata table) {
+    tabletMap.removeTableMappings(table.getKeyspace().getName(), table.getName());
+  }
+
+  @Override
+  public void onKeyspaceRemoved(KeyspaceMetadata keyspace) {
+    // Should be covered by onTableRemoved
```

Comment:
Let's don't guess, it worth only one line of code.

Reply:
Switched to calling `tabletMap.removeTableMappings(keyspace.getName());`